### PR TITLE
Add configuration option to set suffix for service name.

### LINKED ISSUE #195: Make instanceId configurable
This is a follow up on #33. 

Make instanceId configurable so when an application uses both HTTP and gRPC with service discovery, users can define a different name for the gRPC application to register to service discovery.

### PR DESCRIPTION
**Pull request by ilopmar**
Fixes #195

### CODE REVIEW

**Review by alvarosanchez — APPROVED** [review]

**Comment by graemerocher** [comment]
Not sure I agree with this approach. I would call it `grpc.server.instance-id` and allow overridding the entire id registered. Customizing the suffix doesn't make sense to me.

File: src/main/docs/guide/serviceDiscovery.adoc [hunk 1]
```diff
@@ -37,6 +37,23 @@ consul:
...
+----
+micronaut:
+    application:
+        name: greeter # <1>
+grpc:
+    server:
+        instance-id: 'service-grpc' # <2>
+----
+<1> The HTTP port will be registered in Consul with the name `greeter`
+<2> The GRPC port will be registered in Consul with the name `hello-grpc`
```
Comment (graemerocher) [inline src/main/docs/guide/serviceDiscovery.adoc:54]: This documentation appears to be incorrect

File: grpc-server-runtime/src/main/java/io/micronaut/grpc/server/GrpcServerConfiguration.java [hunk 2]
```diff
@@ -118,6 +119,22 @@ public int getServerPort() {
...
+    /**
+     * The instance id.
+     * @return The instance id
+     */
+    public @Nonnull String getInstanceId() {
+        return instanceId;
+    }
+
+    /**
+     * Sets the instance id.
```
Comment (graemerocher) [inline grpc-server-runtime/src/main/java/io/micronaut/grpc/server/GrpcServerConfiguration.java:131]: Need more clarification in this javadoc what the instance ID is used for (ie service discovery)

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: src/main/docs/guide/serviceDiscovery.adoc (hunk 1 above)
Comment (ilopmar) [reply src/main/docs/guide/serviceDiscovery.adoc:54]: Sorry, bad copy-paste.

File: grpc-server-runtime/src/main/java/io/micronaut/grpc/server/GrpcServerConfiguration.java (hunk 2 above)
Comment (ilopmar) [reply grpc-server-runtime/src/main/java/io/micronaut/grpc/server/GrpcServerConfiguration.java]: Done

**Pull request merged**